Code/Tools/FBuild/FBuildCore/Graph/DependencyListNode.cpp

```diff
@@ -0,0 +1,243 @@
+// DependencyListNode.cpp
+//------------------------------------------------------------------------------
+
+// Includes
+//------------------------------------------------------------------------------
+#include "DependencyListNode.h"
+
+#include "Tools/FBuild/FBuildCore/FBuild.h"
+#include "Tools/FBuild/FBuildCore/FLog.h"
+#include "Tools/FBuild/FBuildCore/BFF/Functions/Function.h"
+#include "Tools/FBuild/FBuildCore/Graph/AliasNode.h"
+#include "Tools/FBuild/FBuildCore/Graph/NodeGraph.h"
+
+#include "Core/Containers/Array.h"
+#include "Core/Env/ErrorFormat.h"
+#include "Core/FileIO/FileIO.h"
+#include "Core/FileIO/FileStream.h"
+#include "Core/FileIO/PathUtils.h"
+#include "Core/Strings/AStackString.h"
+
+// REFLECTION
+//------------------------------------------------------------------------------
+REFLECT_NODE_BEGIN( DependencyListNode, Node, MetaNone() )
+    REFLECT(        m_Source,                   "Source",                   MetaFile() )
+    REFLECT(        m_Dest,                     "Dest",                     MetaPath() )
+    REFLECT_ARRAY(  m_Patterns,                 "Patterns",                 MetaOptional() )
+    REFLECT_ARRAY(  m_PreBuildDependencyNames,  "PreBuildDependencies",     MetaOptional() + MetaFile() + MetaAllowNonFile() )
+REFLECT_END( DependencyListNode )
+
+// Anonymous DependencyListBuilder struct
+//------------------------------------------------------------------------------
+struct DependencyListBuilder {
+    enum {
+        iBuckets = 256,
+        mBuckets = (iBuckets - 1)
```

Comment:
This enum is rather confusing, particularly with one of the enumerations using member variable naming style.